File: wordflip-server/src/main/java/com/wordflip/dto/ErrorResponse.java
package com.wordflip.dto;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.time.Instant;
import java.util.Map;

/**
 * Aligns with openapi.yaml components/schemas/ErrorResponse.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record ErrorResponse(
        String code,
        String message,
        Instant timestamp,
        String path,
        Map<String, Object> details
) {
    public static ErrorResponse of(String code, String message, String path) {
        return new ErrorResponse(code, message, Instant.now(), path, null);
    }

    public static ErrorResponse of(String code, String message, String path, Map<String, Object> details) {
        return new ErrorResponse(code, message, Instant.now(), path, details);
    }
}

File: wordflip-android/core-model/src/main/java/com/wordflip/core/model/ErrorResponse.kt
package com.wordflip.core.model

/**
 * 统一错误响应，对齐 openapi ErrorResponse；用于解析 4xx/5xx 响应体。
 */
data class ErrorResponse(
    val code: String,
    val message: String,
    val timestamp: String? = null,
    val path: String? = null,
    val details: Map<String, Any>? = null,
)
